{{cookiecutter.project_slug}}/docker/django/Dockerfile

```diff
@@ -0,0 +1,47 @@
+FROM python:3.6-alpine
+
+ENV PYTHONUNBUFFERED 1
+
+ARG environment
+
+RUN apk update \
+  # psycopg2 dependencies
+  && apk add --virtual build-deps gcc python3-dev musl-dev \
+  && apk add postgresql-dev \
+  # Pillow dependencies
+  && apk add jpeg-dev zlib-dev freetype-dev lcms2-dev openjpeg-dev tiff-dev tk-dev tcl-dev \
+  # CFFI dependencies
+  && apk add libffi-dev py-cffi \
+  && [ "${environment}" != 'production' ] && apk add \
+    # Translations dependencies
+    gettext \
+    # and https://docs.djangoproject.com/en/dev/ref/django-admin/#dbshell
+    postgresql-client \
+    || true
+
+COPY ./requirements/ /app/requirements/
+WORKDIR /app/
+RUN [ "${environment}" = 'production' ] && pip --no-cache-dir install -r "./requirements/${environment}.txt" || pip install -r "./requirements/${environment}.txt"
+
+RUN addgroup -S django \
+    && adduser -S -g django django
+
+COPY ./docker/django/scripts/ /scripts/
+RUN [ "${environment}" = 'production' ] && find /scripts/ -type f -name '*.local' -exec rm -f {} \; || true
+RUN chown --recursive django:django /scripts/
+RUN find /scripts/ -type f -exec sed -i 's/\r//' {} \;
+RUN chmod +x /scripts/*
+RUN mv /scripts/* / \
+    && rmdir /scripts/
+
+COPY . /app/
```

Comment:
I might be missing something, but won't this slow down local development? My understanding was that it's not required locally as we mount the project root under `/app`:

https://github.com/pydanny/cookiecutter-django/blob/cf8287e5763d3057540ac24fe85c422b7d59413c/%7B%7Bcookiecutter.project_slug%7D%7D/local.yml#L19

Reply:
This is one of the two ways I could think about to facilitate Dockerfile reuse across environments: COPYing the code while further mounting the same dir over the container file system won't slow anything down from my experience with a number of production-grade projects :)

Reply:
Speaking of Dockerfile reuse, unforrtunately, it's impossible now unless you use some clever layer caching-reuse which is bashizm at its finest at least, and producing unmaintainable pile of confiuration at worst... 
N.B. HashiCorp's [Packer](https://www.packer.io/) is a real cure for that, I suggest you find time to take a look at it, if you can -- definitely a noteworthy thing.